lib/engine/game/g_18_fr/game.rb

```diff
@@ -0,0 +1,225 @@
+             90p
+             100p
+             110p
+             120s
+             135p
+             150p
+             165p
+             180p
+             200p
+             220
+             245
+             270
+             300
+             330
+             360
+             400
+             440
+             490
+             540
+             600
+             660
+             720
+             780
+             840
+             900],
+           ].freeze
+
+        PHASES = [
+          {
+            name: 'Yellow',
+            train_limit: 4,
+            tiles: [:yellow],
+            operating_rounds: 2,
+            corporation_sizes: [2],
+          },
+          {
+            name: 'Yellow+',
+            on: '2+',
+            train_limit: 4,
+            tiles: [:yellow],
+            operating_rounds: 2,
+            corporation_sizes: [2, 5],
+          },
+          {
+            name: 'Green',
+            on: '3+',
+            train_limit: 4,
+            tiles: %i[yellow green],
+            operating_rounds: 2,
+            corporation_sizes: [2, 5],
+          },
+          {
+            name: 'Blue',
+            on: '3P',
+            train_limit: 3,
+            tiles: %i[yellow green],
+            operating_rounds: 2,
+            corporation_sizes: [2, 5],
+          },
+          {
+            name: 'Brown',
+            on: '5',
+            train_limit: 3,
+            tiles: %i[yellow green brown],
+            operating_rounds: 2,
+            corporation_sizes: [2, 5],
+          },
+          {
+            name: 'Gray',
+            on: '6*D',
+            train_limit: 3,
+            tiles: %i[yellow green brown],
+            operating_rounds: 2,
+            corporation_sizes: [2, 5],
+          },
+        ].freeze
+
+        TRAINS = [{ name: '2', distance: 2, price: 100, rusts_on: '3P', num: 40 },
+                  { name: '2+', distance: 2, price: 100, obsolete_on: '3P', num: 4 },
+                  {
+                    name: '3+',
+                    distance: 3,
+                    price: 300,
+                    obsolete_on: '6*D',
+                    num: 12,
+                    events: [{ 'type' => 'two_tile_lays' }],
+                  },
+                  {
+                    name: '3P',
+                    distance: 3,
+                    price: 400,
+                    num: 1,
+                    events: [{ 'type' => 'free_ports' }],
+                  },
+                  { name: '2P', distance: 2, price: 300, num: 5 },
+                  { name: '5', distance: 5, price: 600, num: 6 },
+                  { name: '6*D', distance: 6, price: 800, num: 30 },
+                  { name: '2P*', distance: 2, price: 200, num: 1 }].freeze
+
+        EVENTS_TEXT = Base::EVENTS_TEXT.merge(
+          'two_tile_lays' => ['Corporations may lay two tiles for 20 F. One may be upgrade'],
+          'free_ports' => ['Ports no longer count towards train length']
```

Comment:
As these are ongoing effects, these might be better implemented as phases statuses. If they are events they will disappear from the game information page once they have happened, but statuses are visible for the whole game.